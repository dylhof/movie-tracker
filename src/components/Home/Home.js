import React from 'react';
import {connect} from 'react-redux'

const Home = (props) => {
  const movieCards = props.movies.map(movie => {
    return <p>{movie.title}</p>
  })
  return (
    <div>{movieCards}</div>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(Home)


// display all movies in store
// favorite funtionality will live on movie card
