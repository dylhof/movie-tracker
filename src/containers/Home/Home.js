import React from 'react'
import { connect } from 'react-redux'
import MovieCard from '../MovieCard/MovieCard'
import PropTypes from 'prop-types'

export const Home = (props) => {
    const movieCards = props.movies.map(movie => {
      return (
        <MovieCard {...movie} key={movie.id} />
      )
    })
    return (
      <div className='home'>{movieCards}</div>
    )
}

export const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps)(Home)

Home.propTypes = {
  movies: PropTypes.array
}