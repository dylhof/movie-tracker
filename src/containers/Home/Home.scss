.home,
.favorites {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr))
}

.add-fave-div {
  height: 50px;
  width: 75%;
  margin: 50px auto;
  background: #0D0707;
  box-shadow: 5px 5px 7px #2b0c0e;
  border-radius: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
  .add-fave-message {
    color: $font-color;
    font-size: 30px;
  }
}

