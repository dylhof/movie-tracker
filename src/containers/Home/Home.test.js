import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';
import { mapStateToProps } from './Home'

describe('home', () => {
  let wrapper
  beforeEach(() => {
    let mockMovies = [{title: 'Aquaman', id: 1}, {title: 'Serenity', id: 2}]
    wrapper = shallow(
      <Home movies={mockMovies} /> 
    )
  })
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    
    it('should return an object with a movies array', () => {
      //setup
      const mockState = {movies: [{}], favorites: [1]}
      const expected = {movies: [{}]}
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })
})