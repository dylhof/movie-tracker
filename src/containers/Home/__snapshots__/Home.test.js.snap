// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
